### PR TITLE
Feat/integration tests refactor

### PR DESCRIPTION
**Pull request by ditchfieldcaleb**

### CODE REVIEW

File: test/integration/tests/staking.ts [hunk 1]
```diff
@@ -0,0 +1,559 @@
...
+import proposals from '@test/integration/proposals_config.json';
+import { TestEndtoEndCoordinator } from '../setup';
+import { forceEth } from '@test/integration/setup/utils';
+
+const toBN = ethers.BigNumber.from;
+
+before(async () => {
+  chai.use(CBN(ethers.BigNumber));
+  chai.use(solidity);
+  await resetFork();
```
Comment (ElliotFriedman) [inline test/integration/tests/staking.ts:19]: Does this before block execute before the top level describe block starts?

File: test/helpers.ts [hunk 2]
```diff
@@ -70,9 +71,19 @@ async function increaseTime(amount: number) {
 }
 
 async function resetTime() {
```
Comment (Joeysantoro) [inline test/helpers.ts:73]: resetTime is a bit unexpected because it also resets state.

File: test/helpers.ts [hunk 3]
```diff
@@ -70,9 +71,19 @@ async function increaseTime(amount: number) {
...
+}
+
+async function resetFork() {
   await hre.network.provider.request({
     method: 'hardhat_reset',
-    params: []
+    params: [
+      {
+        forking: {
+          jsonRpcUrl: hre.config.networks.hardhat.forking.url
```
Comment (Joeysantoro) [inline test/helpers.ts:83]: this will error if forking=false right?

Comment (Joeysantoro) [reply test/helpers.ts:83]: What block does it use?

File: test/integration/tests/staking.ts (hunk 1 above)
Comment (ditchfieldcaleb) [reply test/integration/tests/staking.ts:19]: Yes

File: hardhat.config.ts [hunk 4]
```diff
@@ -53,8 +53,8 @@ export default {
       chainId: 5777, // Any network (default: none)
       forking: enableMainnetForking
         ? {
-            url: `https://eth-mainnet.alchemyapi.io/v2/${mainnetAlchemyApiKey}`,
-            blockNumber: 13398685
+            url: `https://eth-mainnet.alchemyapi.io/v2/${mainnetAlchemyApiKey}`
+            //blockNumber: <set>
```
[the remark this answers is not in the document]
Comment (ditchfieldcaleb) [reply hardhat.config.ts:57]: Not todo - just commented so that you can see how to specify a particular block number if you need it.

**Review by Joeysantoro — APPROVED** [review]

**Pull request merged**